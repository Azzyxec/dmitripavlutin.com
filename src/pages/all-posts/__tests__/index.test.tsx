import { shallow } from 'enzyme';

import PlainListAllFetch from '../index';

const props = {
  pageContext: {
    tag: 'javascript',
  },
  data: {
    allMarkdownRemark: {
      edges: [
        {
          node: {
            frontmatter: {
              title: 'Useful techniques to facilitate React server-side rendering',
              description: 'Useful techniques to facilitate React server-side rendering.',
              published: '2018-03-17',
              modified: '2019-01-01',
              commentsThreadId: 'thread-id',
              slug: 'useful-techniques-react-server-side-rendering',
              tags: ['react', 'server side rendering'],
            },
          },
        },
      ],
    },
  },
};

describe('<PlainListAllFetch />', function() {
  it('should render plain list excerpts template', function() {
    const wrapper = shallow(<PlainListAllFetch {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
