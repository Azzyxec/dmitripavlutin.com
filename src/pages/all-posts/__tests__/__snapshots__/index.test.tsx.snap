// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlainListAllFetch /> should render plain list excerpts template 1`] = `
<PlainListAllTemplate
  posts={
    Array [
      Object {
        "commentsThreadId": "thread-id",
        "description": "Useful techniques to facilitate React server-side rendering.",
        "modified": "2019-01-01",
        "published": "2018-03-17",
        "slug": "useful-techniques-react-server-side-rendering",
        "tags": Array [
          "react",
          "server side rendering",
        ],
        "title": "Useful techniques to facilitate React server-side rendering",
      },
    ]
  }
/>
`;
